java/java.project/src/org/netbeans/api/java/project/JavaProjectConstants.java

```diff
@@ -46,6 +46,13 @@ private JavaProjectConstants() {}
      */
     public static final String SOURCES_TYPE_MODULES = "modules"; // NOI18N
 
+    /**
+     * Generated root sources type.
+     * @see org.netbeans.api.project.Sources
+     * @since 1.88
+     */
+    public static final String SOURCES_TYPE_GENERATED = "generated"; // NOI18N
```

Comment:
Note that other languages may also generate sources; e.g. in Gradle build, one can easily have `build/generated-sources/groovy/...` or kotlin. There's sadly not an is-a formal hierarchy among source types, although Gradle/Groovy support provides one (retursn Groovy source root for "java" query) -  but maybe we could declare "java-resources" from the start here ?

@JaroslavTulach  ?